File: src/routes/appRoutes/appRoutes.go
package appRoutes

import (
	"i9rfs/src/controllers/appControllers"
	"i9rfs/src/middlewares/authMiddlewares"

	"github.com/gofiber/contrib/websocket"
	"github.com/gofiber/fiber/v2"
)

func Route(router fiber.Router) {
	router.Use(authMiddlewares.UserAuth)

	router.Get("/signout")

	router.Use("/rfs", func(c *fiber.Ctx) error {
		if websocket.IsWebSocketUpgrade(c) {
			return c.Next()
		}

		return fiber.ErrUpgradeRequired
	})

	router.Get("/rfs", appControllers.RFSAction)
}

File: src/routes/appRoutes.ts
import express from "express"
import * as authMiddlewares from "../middlewares/authMiddlewares.ts"
import * as appControllers from "../controllers/app/appControllers.ts"
import * as appValidators from "../validators/appValidators.ts"

const router = express.Router()

router.use(authMiddlewares.UserAuth)

router.post(
  "/uploads/authorize",
  ...appValidators.AuthorizeUpload(),
  appControllers.AuthorizeUploadDoc,
  appControllers.AuthorizeUpload
)

router.get("/signout", appControllers.SignoutDoc, appControllers.Signout)

export default router
